src/layers/sequential.rs

```diff
@@ -60,3 +59,46 @@ where
         Tensor::from_shape_vec(output_shape, outputs).unwrap()
     }
 }
+
+impl<D: Dimension> Model for Sequential<D> {
+    fn state_dict(&self) -> IndexMap<String, Vec<f64>> {
+        let mut state_dict: IndexMap<String, Vec<f64>> = IndexMap::new();
+        let trainable_layers = self.layers.iter().filter(|x| x.is_trainable());
+
+        for layer in trainable_layers {
+            let (weight_key, bias_key) = (layer.name() + ".weight", layer.name() + ".bias");
+
+            state_dict.insert(weight_key, layer.weights());
+            state_dict.insert(bias_key, layer.biases());
+        }
+
+        state_dict
+    }
+
+    fn load_state_dict(&mut self, path: &str) {
+        let file = File::open(path).unwrap();
+        let state_dict: IndexMap<String, Vec<f64>> = de::from_reader(file).unwrap();
+
+        let countable_trainable_layers = self.layers.iter().filter(|x| x.is_trainable());
+
+        for layer in countable_trainable_layers {
+            let (weight_key, bias_key) = (layer.name() + ".weight", layer.name() + ".bias");
+
+            if state_dict.get(&weight_key).unwrap().len() != layer.weights().len() {
+                panic!("Wrong loaded weight count for layer \"{}\".", layer.name());
+            }
+            if state_dict.get(&bias_key).unwrap().len() != layer.biases().len() {
+                panic!("Wrong loaded bias count for layer \"{}\".", layer.name());
+            }
+        }
+
+        let trainable_layers = self.layers.iter_mut().filter(|x| x.is_trainable());
+
+        for layer in trainable_layers {
+            let (weight_key, bias_key) = (layer.name() + ".weight", layer.name() + ".bias");
+
+            layer.set_weights(state_dict.get(&weight_key).unwrap().clone());
+            layer.set_biases(state_dict.get(&bias_key).unwrap().clone());
```

Comment:
Can this logic be combined into one for-loop? Also is cloning the value of either the weight key or bias key needed?

```suggestion
        let trainable_layers = self.layers.iter_mut().filter(|x| x.is_trainable());

        for layer in trainable_layers {
            let (weight_key, bias_key) = (layer.name() + ".weight", layer.name() + ".bias");

            let saved_weights = state_dict.get(&weight_key).unwrap();
            if saved_weights.len() != layer.weights().len() {
                panic!("Wrong loaded weight count for layer \"{}\".", layer.name());
            }
            layer.set_weights(saved_weights.clone());
            
            let saved_bias = state_dict.get(&bias_key).unwrap();
            if saved_bias.len() != layer.biases().len() {
                panic!("Wrong loaded bias count for layer \"{}\".", layer.name());
            }
            layer.set_biases(saved_bias.clone());
```

Reply:
You know what, you're right. I was thinking that counting must be done through a separate iterator first, because you could be loading a model halfway and then the counts mismatch, resulting in a corrupted model. However this is not at all an issue when considering that `panic!` will exit the program and drop all the allocated program memory anyway.

Reply:
Also, the clone is necessary because `set_weights` takes ownership of the passed `Vec<f64>`. But changing the signature so that it borrows the `Vec<f64>` instead does eliminate the need for clone, so I'll do that.